File: repositories/clientsRepo.js
let clients = [];
let nextId = 1;

const add = (data) => {
  const client = {
    id: nextId++,
    name: data.name,
    email: data.email,
    phone: data.phone || null,
    created_at: new Date().toISOString()
  };
  clients.push(client);
  return client;
};

const getAll = () => clients.map(c => ({ id: c.id, name: c.name, email: c.email }));

const getById = (id) => clients.find(c => c.id === id) || null;

const update = (id, data) => {
  const c = clients.find(x => x.id === id);
  if (!c) return null;
  if (data.name !== undefined) c.name = data.name;
  if (data.email !== undefined) c.email = data.email;
  if (data.phone !== undefined) c.phone = data.phone;
  return c;
};

module.exports = { add, getAll, getById, update };

File: clients-service/app/repositories/clients_repo.py
class ClientsRepo:
    def __init__(self):
        self.clients = []
        self.next_id = 1

    def add(self, data):
        client = {
            "id": self.next_id,
            "name": data.get("name"),
            "email": data.get("email"),
            "phone": data.get("phone")
        }
        self.clients.append(client)
        self.next_id += 1
        return client

    def get_all(self):
        return self.clients

    def get_by_id(self, id):
        return next((c for c in self.clients if c["id"] == id), None)

    def update(self, id, data):
        client = self.get_by_id(id)
        if not client:
            return None
        if "name" in data: client["name"] = data["name"]
        if "email" in data: client["email"] = data["email"]
        if "phone" in data: client["phone"] = data["phone"]
        return client
